pandas/tests/indexes/test_range.py

```diff
@@ -610,67 +612,73 @@ def test_union(self):
-                              I64(range(20)),
-                              RI(0, -10, -1),
-                              I64(range(0, -20, -1)),
-                              I64(list(range(0, 10, 2)) +
-                                  list(range(1, 10, 2))),
-                              I64(list(range(0, 11, 2)) +
-                                  list(range(1, 12, 2))),
-                              I64(list(range(0, 21, 4)) +
-                                  list(range(-2, 24, 4))),
-                              I64(list(range(0, -20, -2)) +
-                                  list(range(-1, -21, -2))),
-                              I64(range(0, 100, 5)),
-                              I64(list(range(0, -100, -5)) + [5]),
-                              I64(list(range(0, -11, -1)) + [1, -11]),
-                              RI(0),
-                              RI(0, -10, -2),
-                              I64(range(0, 102, 2)),
-                              I64(list(range(0, -100, -2)) + [-100, 2]),
-                              I64(list(range(0, -100, -1))),
-                              I64([0, 5]),
-                              I64([0, 5, -5]),
-                              I64([0, 1, 2, 4]),
-                              RI(0, 10, 1),
-                              I64([1, 5, 6])]
-
-        for ((idx1, idx2), expected) in zip(inputs, expected_notsorted):
+            tm.assert_index_equal(res2, expected_sorted, exact=True)
+            tm.assert_index_equal(res3, expected_sorted)
+
+    def test_union_notsorted(self, union_inputs):
+        RI = RangeIndex
+        I64 = Int64Index
+
+        expected = [RI(0, 10, 1),
+                    I64(range(20)),
+                    I64(range(20)),
+                    RI(0, -10, -1),
+                    I64(range(0, -20, -1)),
+                    I64(list(range(0, 10, 2)) + list(range(1, 10, 2))),
+                    I64(list(range(0, 11, 2)) + list(range(1, 12, 2))),
+                    I64(list(range(0, 21, 4)) + list(range(-2, 24, 4))),
+                    I64(list(range(0, -20, -2)) + list(range(-1, -21, -2))),
+                    I64(range(0, 100, 5)),
+                    I64(list(range(0, -100, -5)) + [5]),
+                    I64(list(range(0, -11, -1)) + [1, -11]),
+                    RI(0),
+                    RI(0, -10, -2),
+                    I64(range(0, 102, 2)),
+                    I64(list(range(0, -100, -2)) + [-100, 2]),
+                    I64(list(range(0, -100, -1))),
+                    I64([0, 5]),
+                    I64([0, 5, -5]),
+                    I64([0, 1, 2, 4]),
+                    RI(0, 10, 1),
+                    I64([1, 5, 6])]
+
+        for ((idx1, idx2), expected_notsorted) in zip(union_inputs, expected):
             res1 = idx1.union(idx2, sort=False)
-            tm.assert_index_equal(res1, expected, exact=True)
+            tm.assert_index_equal(res1, expected_notsorted, exact=True)
```

Comment:
I have tried to make the tests a bit clearer by using a fixture for the input data as you suggested which is shared by the two tests. I could combine the two tests into one and then make a list of tuples of the form ``[(expected_sorted, expected_notsorted) ... ]`` but I found it a bit clearer to have two separate tests as above.

Reply:
not what i meant. make  tuple of (input, expected_sorted, expected_not_sorted) as a single fixture. the fixtures is a list of these tuples.